src/jquery.multiselect.js

```diff
@@ -401,6 +488,38 @@
       // Update saved labels and inputs
       this.$labels = this.$menu.find('label');
       this.$inputs = this.$labels.children('input');
+
+      // If the filter widget is in use, then also update its cache.
+      if ( this.element.is(':data("ech-multiselectfilter")') ) {
+            this.element.multiselectfilter('instance').updateCache(true);
+      }
+    },
+
+    /**
+     * Updates the widget checkboxes' checked states
+     * from the native select options' selected states.
+     * @param {boolean} skipDisabled If true, disabled options in either are skipped.
+     */
+    resync : function(skipDisabled) {
+      var $inputs = this.$inputs;
+      var $options = this.element.find('options');
```

Comment:
`$options` always appears to be 0, which means this just defers to `refresh` every time.

Reply:
Ok... good catch...   It should be `.find('option')`  not  `.find('options')`   Fat fingers got the best of me.  😄

Reply:
It happens!